base_layer/wallet/src/transaction_service/handle.rs

```diff
@@ -150,7 +150,7 @@ pub enum TransactionEvent {
     TransactionDirectSendResult(TxId, bool),
     TransactionCompletedImmediately(TxId),
     TransactionStoreForwardSendResult(TxId, bool),
-    TransactionCancelled(TxId),
+    TransactionCancelled(TxId, u64),
```

Comment:
Rather use the actual enum here and cast it to u64 when it crosses an application boundary
```suggestion
    TransactionCancelled(TxId, TxRejection),
```